### DISCUSSION TITLE
Do we still need to define the type explicitly when it's nullable

### DISCUSSION THREAD

**Discussion by micalevisk** [nartc/mapper#503] [Q&A] [answered]
Using v8.7.6

If I recall correctly, in the past we'd have to use `@AutoMap(() => Type)` if the type of some property has an union with `null`. But now if we use `nullSubstitution` it doesn't maps `null` anymore and I believe this is a bug -- actually, looks like the `nullSubstitution` is lost when we use `extend`

But my question is: do we still need to explicitly type destination's type if it has an union with `null`? I didn't find any mention to that in the docs

<details>
<summary>See demo code</summary>

```ts
import { createMap, createMapper, extend, nullSubstitution, forMember } from '@automapper/core';
import { classes, AutoMap } from '@automapper/classes';

const mapper = createMapper({
  strategyInitializer: classes(),
});

class S {
  @AutoMap(() => String) // Doesn't work
  // @AutoMap() // Does works
  foo: string | null
}

class D {
  @AutoMap()
  foo?: string
}

class D2 extends D {}

createMap(mapper,
  S, D,
  forMember(
    d => d.foo,
    nullSubstitution(undefined),
  )
)
createMap(mapper,
  S, D2,
  extend(S, D)
)


const s = new S()
s.foo = null

console.log(
  mapper.map(s, S, D) ,
  mapper.map(s, S, D2) ,
)
/*
D { foo: undefined }
D2 { foo: null }
*/
```

</details>

**Comment by micalevisk** [comment]
ping @nartc :cry:

**Reply by nartc** [reply]
Sorry for the delay, I'm currently on vacation (have been on vacation for 2 months) and will be back in about 3 weeks.

**Comment by nartc** [comment, marked as answer]
Hi @micalevisk, it was a bug with `extend()`. I kept the old behavior (you still need to explicitly set a type for Null union). Check out the latest version
